src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp

```diff
@@ -87,13 +87,20 @@ namespace
             return Model;
         }
 
-        virtual void prepare_inputs(void* data) const APPLESEED_OVERRIDE
+        virtual void prepare_inputs(
+            const ShadingPoint& shading_point,
```

Comment:
prepare inputs is independent of incoming or outgoing points.
There's only one shading point, the point at which you are going to 
evaluate the bssrdf.